test/unit/webui/components/search.spec.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * @flow
+ * @prettier
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import Search from '../../../../src/webui/components/Search/index';
+
+const SEARCH_FILE_PATH = '../../../../src/webui/components/Search/index';
+
+// Global mocks
+const event = {
+  stopPropagation: jest.fn(),
+};
+
+window.location.assign = jest.fn();
+
+describe('<Search /> component test', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = mount(<Search />);
+  });
+
+  test('should load the component in default state', () => {
+    expect(wrapper.html()).toMatchSnapshot();
+  });
+
+  test('onBlur: should cancel all search requests', async () => {
+    const { onBlur, requestList } = wrapper.instance();
+    const spy = jest.spyOn(wrapper.instance(), 'cancelAllSearchRequests');
+    // add one request
+    const request = {
+      abort: jest.fn(),
+    };
+    wrapper.instance().requestList = [request];
```

Comment:
I'd love to know more about this hacky 👨‍💻 line. 🤓

Reply:
It adds a cancellable `request` to `requestList`